lib/rules/no-global-setup.js

```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+/**
+ * @fileoverview Disallow using setup and teardown methods outside a suite
+ * @author Omer Ganim
+ */
+
+var suiteRegexp = /^(f|d|x)?describe$/
+var setupRegexp = /^(before|after)(Each|All)$/
+
+module.exports = function (context) {
+  var suiteDepth = 0
+  return {
+    CallExpression: function (node) {
+      if (suiteRegexp.test(node.callee.name)) {
+        suiteDepth++
```

Comment:
interesting approach for determining the context; I wonder if this is more performant than this approach https://github.com/tlvince/eslint-plugin-jasmine/blob/master/lib/rules/no-unsafe-spy.js#L10-L18

NB: I'm a little concerned about the reliance on the details of how the linter traverses the tree.